File: apps/frontend/src/types/analyst.ts
export interface FindingItem {
  finding: string;
  severity: 'CRITICAL' | 'HIGH' | 'MEDIUM' | 'LOW' | 'INFO';
  rule_id?: string;
  evidence_ids: string[];
}

export interface PossibleCauseItem {
  hypothesis: string;
  confidence: 'HIGH' | 'MEDIUM' | 'LOW';
  basis: string;
  evidence_ids: string[];
}

export interface RecommendedActionItem {
  action: string;
  owner_role: 'CEO' | 'GENERAL_MANAGER' | 'FLOOR_MANAGER' | 'KITCHEN_LEAD';
  priority: 'CRITICAL' | 'HIGH' | 'MEDIUM' | 'LOW';
  approval_required: boolean;
  evidence_ids: string[];
}

export interface DecisionAuditLogItem {
  log_id: string;
  brief_id: string;
  decision_id?: string;
  previous_status: string;
  new_status: string;
  actor_role: string;
  action_type: string;
  timestamp: string;
  comment?: string;
}

export interface AnalystBriefResponse {
  brief_id: string;
  business_date: string;
  dataset_type: string;
  status: 'READY' | 'BLOCKED' | 'REVIEW_REQUIRED' | 'APPROVED' | 'REJECTED';
  provider: string;
  model: string;
  prompt_version: string;
  facts_version: string;
  rule_version: string;
  executive_summary: string;
  findings: FindingItem[];
  possible_causes: PossibleCauseItem[];
  recommended_actions: RecommendedActionItem[];
  unknowns: string[];
  evidence_ids: string[];
  rejection_reasons: string[];
  created_at: string;
  reviewed_at?: string;
  approval_disclaimer: string;
}

File: apps/api/schemas/analyst.py
# -*- coding: utf-8 -*-
from typing import List, Dict, Any, Optional, Literal
from pydantic import BaseModel, Field
from domains.analyst.schemas import FindingItem, PossibleCauseItem, RecommendedActionItem

class AnalystBriefResponse(BaseModel):
    brief_id: str
    business_date: str
    dataset_type: str = "SYNTHETIC"
    status: Literal['READY', 'BLOCKED', 'REVIEW_REQUIRED', 'APPROVED', 'REJECTED', 'PROVIDER_UNAVAILABLE', 'ERROR']
    
    requested_provider: str = "mock"
    actual_provider: str = "mock"
    fallback_used: bool = False
    fallback_reason: Optional[str] = None
    provider: str
    model: str
    
    prompt_version: str
    facts_version: str
    rule_version: str
    validator_version: str = "v1.1"
    provider_version: str = "v1.1"
    code_version: str = "1.1.0"
    dataset_sha256: Optional[str] = None
    
    executive_summary: str
    findings: List[FindingItem] = Field(default_factory=list)
    possible_causes: List[PossibleCauseItem] = Field(default_factory=list)
    recommended_actions: List[RecommendedActionItem] = Field(default_factory=list)
    unknowns: List[str] = Field(default_factory=list)
    evidence_ids: List[str] = Field(default_factory=list)
    rejection_reasons: List[str] = Field(default_factory=list)
    created_at: str
    reviewed_at: Optional[str] = None
    
    review_authentication_status: str = "SIMULATED"
    identity_verified: bool = False
    approval_disclaimer: str = "DEVELOPMENT HUMAN APPROVAL SIMULATION"

class DecisionAuditLogItem(BaseModel):
    log_id: str
    brief_id: str
    decision_id: Optional[str] = None
    previous_status: str
    new_status: str
    actor_role: str
    action_type: str
    timestamp: str
    comment: Optional[str] = None
    previous_hash: str
    event_hash: str
    review_authentication_status: str = "SIMULATED"

class HumanReviewActionRequest(BaseModel):
    reviewer_role: Literal['CEO', 'GENERAL_MANAGER'] = 'CEO'
    comment: Optional[str] = None

